packages/core/types/src/pricing/common/price-set.ts

```diff
@@ -206,6 +207,11 @@ export interface CalculatedPriceSet {
   }
 }
 
+export interface RuleWithOperator {
+  operator: PricingRuleOperatorValues
+  value: number
```

Comment:
Somewhat related to my other comment, I wonder if we should let this be more open-ended at the module level and enforce the type higher in the stack.

It makes sense right now, because of the operators we support, but if we extend that to have `in` and `nin`, then we'd need to be more flexible with the type of value. 

Alternatively, we just make this always be a string and ensure to cast it to a number whenever that is needed

Reply:
I'll answer both comments:

There's a catch here - our pricing calculation is on a SQL level, the query we build does transformation that converts a string to a numeric on a db level. If that value that we convert is not an actual number, the entire price calculation would break everytime there is incompatible data in any of the results during calculation. 

We could add some checks on a sql level, but it doesn't look pretty. The module would have to safely store the data for the operators it supports. Thats why I took a strict-first approach. Good news is, to support additional operators, we just have to add input validations on the service and add a conditional in the query building.

I should've added this on the description. 🤞🏻 😄 wdyt?

Reply:
Yeah, it does make sense, and I think this is a solid first iteration. My questions were mostly out of curiosity about how we will extend this in the future.

When we introduce new operation, I guess we would also need to include some sort of mapping between what type a value can have for a specific operator. For example, you won't use a string value for `gte`, etc.

Reply:
yes, exactly. Shall we merge this as is then?

Reply:
Yes let's do it!